packages/graph-engine/src/nodes/string/case.ts

```diff
@@ -0,0 +1,100 @@
+import { INodeDefinition, ToInput, ToOutput } from '../../index.js';
+import { Node } from '../../programmatic/node.js';
+import { StringSchema } from '../../schemas/index.js';
+
+export enum CaseType {
+	CAMEL = 'camel',
+	SNAKE = 'snake',
+	KEBAB = 'kebab',
+	PASCAL = 'pascal'
+}
+
+/**
+ * This node converts strings between different case formats
+ */
+export default class NodeDefinition extends Node {
+	static title = 'Case Convert';
+	static type = 'studio.tokens.string.case';
+	static description = 'Converts strings between different case formats';
+
+	declare inputs: ToInput<{
+		string: string;
+		type: CaseType;
+		/**
+		 * Characters to be replaced with spaces. Default: -_.
+		 * @default "-_."
+		 */
+		delimiters: string;
+	}>;
+	declare outputs: ToOutput<{
+		string: string;
+	}>;
+
+	constructor(props: INodeDefinition) {
+		super(props);
+		this.addInput('string', {
+			type: StringSchema
+		});
+		this.addInput('type', {
+			type: {
+				...StringSchema,
+				enum: Object.values(CaseType),
+				default: CaseType.CAMEL
+			}
+		});
+		this.addInput('delimiters', {
+			type: {
+				...StringSchema,
+				default: '-_.'
+			}
+		});
+		this.addOutput('string', {
+			type: StringSchema
+		});
+	}
+
+	execute(): void | Promise<void> {
+		const { string, type, delimiters } = this.getAllInputs();
+
+		// Create a RegExp from the delimiters string
+		const delimiterRegex = new RegExp(`[${delimiters}]`, 'g');
+
+		// First normalize the string by splitting on word boundaries
+		const words = string
+			// Add space before capitals in camelCase/PascalCase, but handle consecutive capitals
+			.replace(/([A-Z]+)([A-Z][a-z])/g, '$1 $2')
+			.replace(/([a-z\d])([A-Z])/g, '$1 $2')
+			// Replace delimiters with spaces using the custom regex
+			.replace(delimiterRegex, ' ')
```

Comment:
This is will break as it will only replace the first occurence, so we can rather use replaceAll